File: nodejs_package/tests/brainflow_get_data.ts
import {BoardIds, BoardShim} from 'brainflow';

function sleep (ms: number)
{
    return new Promise ((resolve) => { setTimeout (resolve, ms); });
}

async function runExample (): Promise<void>
{
    const board = new BoardShim (BoardIds.SYNTHETIC_BOARD, {});
    board.prepareSession();
    board.startStream();
    await sleep (3000);
    board.stopStream();
    const data = board.getBoardData();
    board.releaseSession();
    console.info('Data');
    console.info(data);
}

runExample ();

File: swift_package/examples/tests/brainflow_get_data/brainflow_get_data.swift
import BrainFlow
import Foundation

let board = try BoardShim(board_id: BoardIds.SYNTHETIC_BOARD)
try board.prepare_session()
try board.start_stream(buffer_size: 45000)
Thread.sleep(forTimeInterval: 5.0)
try board.stop_stream()
let data = try board.get_board_data()
try board.release_session()

print("Rows: \(data.count)")
print("Samples: \(data.first?.count ?? 0)")
